File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    same = 0
    diff = 0
    first = ''

    for c in s:
        if same == 0:
            first = c
            same = 1
            diff = 0
        else:
            if c == first:
                same += 1
            else:
                diff += 1

        if same == diff:
            answer += 1
            same = 0
            diff = 0

    if same != 0 or diff != 0:
        answer += 1

    return answer

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.kt
class Solution {
    fun solution(s: String): Int {
        var answer = 0
        var sameCount = 0
        var diffCount = 0
        var firstChar: Char? = null

        for (c in s) {
            if (sameCount == 0) {
                firstChar = c
                sameCount = 1
                diffCount = 0
            } else {
                if (c == firstChar) {
                    sameCount++
                } else {
                    diffCount++
                }
            }

            if (sameCount == diffCount) {
                answer++
                sameCount = 0
                diffCount = 0
            }
        }

        if (sameCount != 0 || diffCount != 0) {
            answer++
        }

        return answer
    }
}
